quality_check/10-test_compendium_PLIER.Rmd

```diff
@@ -0,0 +1,308 @@
+#                          ref = 'a2d4a2aa343f9ed4b9b945c04326bebd31533d4d',
+#                          dependencies = TRUE)
+# install.packages("cowplot")
+```
+
+```{r}
+library(PLIER)
+library(org.Dr.eg.db)
+```
+
+#### Directories
+
+```{r}
+pathway_dir <- file.path("data", "pathways")
+result_dir <- "results"
+plot_dir <- "plots"
+```
+
+Results directory specifically for holding the PLIER models
+
+```{r}
+plier_result_dir <- file.path(result_dir, "PLIER_models")
+dir.create(plier_result_dir, showWarnings = FALSE, recursive = TRUE)
+```
+
+## Read in data
+
+### Prior information matrices
+
+We need to get the 3 pathway prior information matrices we have: developmental
+stage, tissues (based on adult only), and the wikipathways into a single 
+matrix.
+We can do this with the `PLIER::combinePaths` function.
+
+```{r}
+# read in the three individual matrices we prepped in 
+# 09-prior_information_matrix
+wikipathway_mat <- readRDS(file.path(pathway_dir, 
+                                     "Danio_rerio_wikipathways_prior_info.RDS"))
+adult_tissue_mat <- readRDS(file.path(pathway_dir,
+                                      "ZFIN_adult_tissue_prior_info.RDS"))
+dev_stage_mat <- readRDS(file.path(pathway_dir,
+                                   "ZFIN_stages_prior_info.RDS"))
+# combine into a single matrix for use with PLIER
+combine_path_mat <- PLIER::combinePaths(adult_tissue_mat,
+                                        dev_stage_mat,
+                                        wikipathway_mat)
+# we no longer need the individual matrices
+rm(adult_tissue_mat, dev_stage_mat, wikipathway_mat)
+```
+
+### Test compendium
+
+We need to convert the test compendium gene identifiers from Ensembl gene IDs to
+ZFIN identifiers so that the IDs in the prior information and the expression
+data match.
+
+```{r}
+# first we need to read in the 
+exprs_file <- file.path("data", "larger_compendium_exprs_mat.RDS")
+exprs_mat <- readRDS(exprs_file)
+```
+
+We expect that there will be a 1:many mapping here, so we'll use the option
+where only the first thing that comes back is returned by setting 
+`multiVals = "first"` (note that this is the default behavior).
+
+```{r}
+# ENSDARG -> ZDB-GENE
+zfin_ids <- mapIds(org.Dr.eg.db, keys = rownames(exprs_mat),
+                   column = "ZFIN", keytype = "ENSEMBL",
+                   multiVals = "first")
+
+# these should be in the same order, but out of an abundance of caution!
+if (all.equal(rownames(exprs_mat), names(zfin_ids))) {
+  rownames(exprs_mat) <- zfin_ids
+} else {
+  stop("Something went wrong!")
+}
+```
+
+## PLIER
+
+We're going to do five repeats.
+
+```{r}
+set.seed(123)
+seeds <- sample(1:100000, 5)
+names(seeds) <- seeds
+```
+
+### Both technologies
+
+```{r}
+# find the common genes between the expression matrix and the prior information
+# matrix
+common_genes <- PLIER::commonRows(exprs_mat, combine_path_mat)
+
+# this expression matrix includes samples from both technologies
```

Comment:
It looks like you are essentially repeating this same code chunk three times but changing just the matrix. Is there a way to streamline this? Maybe make it a function and `lapply`?

Reply:
Writing a function is a good idea. I don't see an advantage to using `lapply` here over calling the function 3 times. I think it will be most convenient to save each of the lists of models separately.

I think you are after getting `coverage_df` through a series of nested `lapply` and then melting the nested list (https://github.com/AlexsLemonade/compendium-processing/pull/22#discussion_r263803682). If it was more than 3 conditions, I would probably go this route. I will instead write the bit getting the three coverage `data.frame` into a function.